CodePush.js

```diff
@@ -204,105 +179,95 @@ function sync(options = {}, syncStatusChangeCallback, downloadProgressCallback)
     
   downloadProgressCallback = typeof downloadProgressCallback == "function" 
     ? downloadProgressCallback 
-    : function(downloadProgress) {
+    : (downloadProgress) => {
         log(`Expecting ${downloadProgress.totalBytes} bytes, received ${downloadProgress.receivedBytes} bytes.`);
       };
   
-  return new Promise((resolve, reject) => {
-    var rejectPromise = (error) => {
-      syncStatusChangeCallback(CodePush.SyncStatus.UNKNOWN_ERROR);
-      log(error.message); 
-      reject(error);
+  try {
+    await CodePush.notifyApplicationReady();
+    
+    syncStatusChangeCallback(CodePush.SyncStatus.CHECKING_FOR_UPDATE);
+    let remotePackage = await checkForUpdate(syncOptions.deploymentKey);
+    
+    let doDownloadAndInstall = async () => {
+      syncStatusChangeCallback(CodePush.SyncStatus.DOWNLOADING_PACKAGE);
+      let localPackage = await remotePackage.download(downloadProgressCallback);
+      
+      syncStatusChangeCallback(CodePush.SyncStatus.INSTALLING_UPDATE);
+      await localPackage.install(syncOptions.installMode, () => {
+        syncStatusChangeCallback(CodePush.SyncStatus.UPDATE_INSTALLED);
+      });
+      
+      return CodePush.SyncStatus.UPDATE_INSTALLED;
     };
     
-    CodePush.notifyApplicationReady()
-      .then(() => {
-        syncStatusChangeCallback(CodePush.SyncStatus.CHECKING_FOR_UPDATE);
-        return checkForUpdate(syncOptions.deploymentKey);
-      })
-      .then((remotePackage) => {
-        var doDownloadAndInstall = () => {
-          syncStatusChangeCallback(CodePush.SyncStatus.DOWNLOADING_PACKAGE);
-          remotePackage.download(downloadProgressCallback)
-            .then((localPackage) => {
-              syncStatusChangeCallback(CodePush.SyncStatus.INSTALLING_UPDATE);
-              return localPackage.install(syncOptions.installMode, () => {
-                syncStatusChangeCallback(CodePush.SyncStatus.UPDATE_INSTALLED);
-                resolve(CodePush.SyncStatus.UPDATE_INSTALLED);
-              });
-            })
-            .catch(rejectPromise)
-            .done();
-        }
+    if (!remotePackage || remotePackage.failedInstall && syncOptions.ignoreFailedUpdates) {
+      syncStatusChangeCallback(CodePush.SyncStatus.UP_TO_DATE);
+      return (CodePush.SyncStatus.UP_TO_DATE);
```

Comment:
I was suggesting removing the parens from the return statement :) I think we should leave them in the second expression of the if statement above.